hexrd/ui/image_load_manager.py

```diff
@@ -40,70 +41,80 @@ def __init__(self):
         super(ImageLoadManager, self).__init__(None)
         self.unaggregated_images = None
 
-    def check_images(self, fnames):
+    def load_images(self, fnames):
+        files = self.explict_selection(fnames)
+        manual = False
+        if not files:
+            files = self.match_files(fnames)
+            matched = self.check_success(files)
+            if not matched:
+                manual = True
+                files = [[fname] for fname in fnames]
+        return files, manual
+
+    def check_success(self, files):
+        detectors = HexrdConfig().detector_names
+        # Make sure there are the same number of files for each detector
+        # and at least one file per detector
+        if (not files[0]
+                or len(files) != len(detectors)
+                or any(len(files[0]) != len(elem) for elem in files)):
+            return False
+        # If the files do not contain detector names they will need to
+        # be manually matched
+        if not all(any(d in f for d in detectors) for f in files[0]):
+            return False
+        return True
```

Comment:
I think we can now simplify this to:

`return all(any(d in f for d in detectors) for f in files[0])`

Reply:
Sounds good - fixed.